include/irr/asset/format/decodePixels.h

```diff
@@ -673,8 +673,8 @@ namespace irr { namespace video // why is this in the video namespace!?
     inline void decodePixels<asset::EF_R16G16_USCALED, double>(const void* _pix[4], double* _output, uint32_t _blockX, uint32_t _blockY)
     {
         const uint32_t& pix = reinterpret_cast<const uint32_t*>(_pix[0])[0];
-        _output[0] = ((pix >> 0) & 0xffffULL);
-        _output[1] = ((pix >> 16) & 0xffffULL);
+        _output[0] = static_cast<double>((pix >> 0) & 0xffffULL);
```

Comment:
@AnastaZIuk the compiler emits a fuckton of warnings about loosing precision, need some "well thought out" `static_cast`s

Unfortunately this can't be done with a script, because sometimes you're not supposed to be casting.